groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/SpockMethodParamCompletion.java

```diff
@@ -0,0 +1,149 @@
+import org.codehaus.groovy.ast.expr.Expression;
+import org.netbeans.modules.csl.api.CompletionProposal;
+import org.netbeans.modules.groovy.editor.api.completion.CaretLocation;
+import org.netbeans.modules.groovy.editor.api.completion.CompletionItem;
+import org.netbeans.modules.groovy.editor.api.completion.util.CompletionContext;
+
+/**
+ *
+ * @author Petr Pisl
+ */
+public class SpockMethodParamCompletion  extends BaseCompletion {
+    
+    private static class SpockParameter implements Variable {
+
+        private final String name;
+
+        public SpockParameter(String name) {
+            this.name = name;
+        }
+        
+        @Override
+        public ClassNode getType() {
+            return ClassHelper.DYNAMIC_TYPE;
+        }
+
+        @Override
+        public ClassNode getOriginType() {
+            return ClassHelper.GROOVY_OBJECT_TYPE;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public Expression getInitialExpression() {
+            return null;
+        }
+
+        @Override
+        public boolean hasInitialExpression() {
+            return false;
+        }
+
+        @Override
+        public boolean isInStaticContext() {
+            return false;
+        }
+
+        @Override
+        public boolean isDynamicTyped() {
+            return true;
+        }
+
+        @Override
+        public boolean isClosureSharedVariable() {
+            return false;
+        }
+
+        @Override
+        public void setClosureSharedVariable(boolean bln) {
+            
+        }
+
+        @Override
+        public int getModifiers() {
+            return 0;
+        }
+        
+    }
+    
+    @Override
+    public boolean complete(Map<Object, CompletionProposal> proposals, CompletionContext request, int anchor) {
+        boolean updated = false;
+        if ((request.location == CaretLocation.INSIDE_METHOD 
+                || request.location == CaretLocation.INSIDE_CLOSURE) 
+                && SpockUtils.isInSpecificationClass(request)) {
+            
+            ListIterator<ASTNode> leafToRoot = request.path.leafToRoot();
+            
+            MethodNode methodNode = null;
+            while(leafToRoot.hasNext()) {
+                ASTNode next = leafToRoot.next();
+                if (next instanceof MethodNode) {
+                    methodNode = (MethodNode) next;
+                    break;
+                }    
+            }
+            String name = null;
+            if (methodNode != null) {
+                name = methodNode.getName();
+            }
+            
+            if (name != null && name.contains("#")) {   //NOI18N
+                String[] parts = name.split("#");       //NOI18N
+                
+                for (int i = 1; i < parts.length; i++) {
+                    String part = parts[i].trim();
+                    if (!part.isEmpty()) {
```

Comment:
Since the `name` is already matched by a regex in split() ... wouldn't it be faster to iterate using `Pattern.compile("#\\p{Alpha}[\\p{Alnum}$_]*)[. ]?*).matcher(name)` ?